### PR TITLE
fix submitted by Zibibbo84 for proxy error

### PR DESCRIPTION
**Pull request by sliptonic**
This patch was submitted via email with very little context.  This is not my code and I have not tested it thoroughly. 
I believe the patch is valid and would like feedback before merging

- [X]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by adrianinsaval** [comment]
if possible use a more meaningful commit message

File: src/Mod/Path/Path/Post/scripts/heidenhain_post.py [hunk 1]
```diff
@@ -344,88 +344,91 @@ def export(objectslist, filename, argstring):
...
-                    obj.recompute()
-                    commands = PathUtils.getPathWithPlacement(obj).Commands
-                    # small edges could be skipped and movements joints can add edges
-                    NameStr = ""
-                    if hasattr(obj, "Label"):
-                        NameStr = str(obj.Label)
-                    if len(commands) != len(STORED_COMPENSATED_OBJ):
-                        # not same number of edges
-                        obj.UseComp = True
+        if hasattr(obj, "Proxy"):
```
Comment (benj5378) [inline src/Mod/Path/Path/Post/scripts/heidenhain_post.py:347]: Avoid nesting
```suggestion
        if not hasattr(obj, "Proxy"):
            continue
```

File: src/Mod/Path/Path/Post/scripts/heidenhain_post.py [hunk 2]
```diff
@@ -344,88 +344,91 @@ def export(objectslist, filename, argstring):
...
+                Object_Kind = "FACE"
+                if LBLIZE_ACTIVE:
+                    LBLIZE_STAUS = True
+            elif isinstance(obj.Proxy, Path.Op.Helix.ObjectHelix):
+                Object_Kind = "HELIX"
+
+            commands = PathUtils.getPathWithPlacement(obj).Commands
+
+            # If used compensated path, store, recompute and diff when asked
+            if hasattr(obj, "UseComp") and SOLVE_COMPENSATION_ACTIVE:
```
Comment (benj5378) [inline src/Mod/Path/Path/Post/scripts/heidenhain_post.py:369]:
```suggestion
            if not hasattr(obj, "UseComp") or not SOLVE_COMPENSATION_ACTIVE:
                continue
```

**Review by benj5378 — COMMENTED** [review]
I generally avoid nesting.
And I would really recommend avoiding this much nesting. Use guard clauses.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9769">pipeline status</a> for feature branch [PR_9769](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9769). Pipeline [903575476](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/903575476) was triggered at [74c6580](https://github.com/FreeCAD/FreeCAD/pull/9769/commits/74c6580c6d3bba1be798826688224aad9b034686). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9769">pipeline status</a> for feature branch [PR_9769](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9769). Pipeline [904516524](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/904516524) was triggered at [6da167a](https://github.com/FreeCAD/FreeCAD/pull/9769/commits/6da167a09899a9f1757223e389b5bf813b1a9332). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/Path/Post/scripts/heidenhain_post.py [hunk 3]
```diff
@@ -344,88 +344,94 @@ def export(objectslist, filename, argstring):
...
+                    LBLIZE_STAUS = True
+            elif isinstance(obj.Proxy, Path.Op.Helix.ObjectHelix):
+                Object_Kind = "HELIX"
+
+            commands = PathUtils.getPathWithPlacement(obj).Commands
+
+            # If used compensated path, store, recompute and diff when asked
+            if not hasattr(obj, "UseComp") or not SOLVE_COMPENSATION_ACTIVE:
+                continue
+                if not obj.UseComp:
```
[the remark this answers is not in the document]
Comment (sliptonic) [reply src/Mod/Path/Path/Post/scripts/heidenhain_post.py:372]: Today I learned:   You have to be careful when accepting code suggestions via github.

**Comment by benj5378** [comment]
So much better, it's 3 indentations less (and that is a lot)!

**Pull request merged**